Improve "No Filter(s) Applied" experience in Overview (CRASM-2279)

hawkishpolicy opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #
- Improve the "No Filter(s) Applied" experience in the Overview page via restoration of a user's initial filters if they are empty when navigating to the Overview page.
- Global Admin can remove all filters via FilterTags in Overview and Inventory, as well as manually through the FilterDrawer component.
   - If this is the case, and a user relies on a client side navigation link, they return a blank page due to a 404 error with no additional feedback. 

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->
- Prevent the display of a No Filters tag in Overview.
- Refactored a useEffect to restore initial filters if empty when navigating to the Overview page.
- Edited nav links in Footer to be client side instead of server side.
- Removed delete icons from disabled filter tags.
- Added Reset Filters button to Inventory if No Results are shown.
- Edited fetchStats useCallback to return if user is globalAdmin and region filters are empty.
- Edited regions call to include an omitted check for org membership alongside globalAdmin status.
   - This was causing an erroneous 403 error.
## 💭 Motivation and context ##
- Closes CRASM-2279
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##
- tested locally.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->


## 📷 Screenshots (if appropriate) ##


https://github.com/user-attachments/assets/47c8124e-0775-4441-93a2-fc97fca5f351


https://github.com/user-attachments/assets/42023c7e-bd81-4092-88e7-22e7d648ad32




## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release.

hawkishpolicy committed:
Added No Results Found message in Risk page

- informs the user that no results were found and to adjust their filters.

hawkishpolicy committed:
Built new context for filter drawer

- Created FilterDrawerContext.ts
- Created FilterDrawerContextProvider.tsx
- Added FilterDrawerContextProvider to App.tsx
- Filter drawer now uses the context to toggle the drawer open and closed
- This was needed as it was not possible to pass the open/closed state of the drawer from the Layout component to the Risk component.

hawkishpolicy committed:
Remove unused import

hawkishpolicy committed:
Update layout.spec.tsx

hawkishpolicy committed:
Refactor to Risk, FilterTags, and Footer

- Removed No Results and Adjust Filter bitton from Risk.
- Brought in initial filters from Layout and restore them if filters array is empty.
- No Results filter tag only displays in Inventory.
- Changed some Footer links to client side routing.

hawkishpolicy committed:
Update snaphot

hawkishpolicy committed:
Update snaphot and edit Link props

- Imported react-router-dom Link component to use with MUI Link component.
- Updated test snaphot.

hawkishpolicy committed:
Remove userLevel code & X from disabled filtertags

- Replaced userLevel logic with useUserLevel hook in Layout and Risk components.
- Removed delete icon from disabled filtertags in FilterTags component.

hawkishpolicy committed:
Add Reset Filter button to Inventory

hawkishpolicy committed:
Merge remote-tracking branch 'origin/develop' into No-Filter-Applied-tag-remains-CRASM-2279

lwersiy reviewed (approved):
Per PR review session 3/19/25 and local testing.
LGTM

ameliav commented:
This is working better than it was before.
However,
- I suggest a TODO to fix the back-end 404 error when switching from the Inventory tab to the Overview tab when no filters are selected.
- I also suggest finding out if there should always be one region and one organization populated in Overview (because that is essentially what's being coded now) and if so, have a TODO to develop the filter bubbles based around that.

ameliav reviewed (changes requested):
This is working better than it was before.
However,
- I suggest a TODO to fix the back-end 404 error when switching from the Inventory tab to the Overview tab when no filters are selected.
- I also suggest finding out if there should always be one region and one organization populated in Overview (because that is essentially what's being coded now) and if so, have a TODO to develop the filter bubbles based around that.

hawkishpolicy committed:
Edit fetchStats call to handle empty region filter

- Will prevent fetchStats call from running if user is global admin and no region filter is applied.
  - This prevents a 404 error.

ameliav reviewed (approved):
404 error is gone now 👍

hawkishpolicy committed:
Add get_org_membership method to organization.py

- regions call will now check for globalAdmin status or Org membership before returning regions.
- fixes an erroneous 403 error.
